wdn/templates_4.1/scripts/wdn-ui.js

```diff
@@ -35,28 +39,47 @@ define(['jquery', 'modernizr'], function($, Modernizr) {
 			// https://developer.apple.com/library/ios/documentation/AppleApplications/Reference/SafariWebContent/pinnedTabs/pinnedTabs.html
 			$('link[rel="mask-icon"]').attr('color', '#d00000');
 
-			this.setUpDropDownWidget('.wdn-dropdown-widget-toggle');
+			this.setUpDropDownWidget('.'+dropdownButtonClass);
 
 			//Close search on escape
 			$(document).on('keydown', function(e) {
 				if (e.keyCode === 27) {
 					//Close on escape
-					closeDropDown('.wdn-dropdown-widget-toggle');
+					closeDropDown('.'+dropdownButtonClass);
 				}
 			});
 
 			//listen for clicks on the document and close dropdowns if they don't come from a dropdown
 			$(document).on('click', function(e) {
-				var $target = $(e.target);
+				//Determine if we need to do anything with our dropdown
+				var $control = $(e.target);
 
-				//dont close this way if the control was clicked
-				if ($target.filter('.wdn-dropdown-widget-toggle, .wdn-dropdown-widget-label').length) {
-					return;
+				//Try to get the control element
+				if ($control.parent('.'+dropdownButtonClass).length) {
+					$control = $control.parent('.'+dropdownButtonClass).first();
+				}
+
+				if ($control.hasClass(dropdownButtonClass)) {
+					var container_id = $control.attr('aria-controls');
+					var $container = $('#' + container_id);
+
+					var isPressed = $control.attr('aria-pressed');
+					if ('true' == isPressed) {
+						$container.attr('aria-hidden', 'true');
+						$control.attr('aria-pressed', 'false');
+					} else {
+						$container.attr('aria-hidden', 'false');
+						$control.attr('aria-pressed', 'true');
+						$container.attr('tabindex', '-1').focus();
+					}
+
+					//Close other widgets
+					closeDropDown($('.'+dropdownButtonClass).not($control));
 				}
 
 				//close all dropdown widgets
-				if (!$('.wdn-dropdown-widget-content').find(e.target).length) {
-					closeDropDown('.wdn-dropdown-widget-toggle');
+				if (!$('.'+dropdownContentClass).find(e.target).length) {
```

Comment:
Where we can, we should reduce the number of times to go to the selector engine with just a class.